code/_onclick/click.dm

```diff
@@ -287,7 +286,17 @@
 	Middle click
 	Only used for swapping hands
 */
-/mob/proc/MiddleClickOn(var/atom/A)
+/mob/proc/MiddleClickOn(var/atom/A, var/params)
+	var/datum/stack/click_handlers
+
+	click_handlers = src.GetClickHandlers()
+
+
+	while (click_handlers.Num())
```

Comment:
While i'm not at all familiar with click handlers on bay, having a while loop every time someone middle clicks seems odd..
Could you convert this to signals / bay's equivalent instead?